packages/date-picker/src/vaadin-date-picker-mixin.js

```diff
@@ -203,24 +219,26 @@ export const DatePickerMixin = (subclass) =>
               firstDayOfWeek: 0,
               today: 'Today',
               cancel: 'Cancel',
-              formatDate: (d) => {
+              referenceDate: '',
+              formatDate(d) {
                 const yearStr = String(d.year).replace(/\d+/, (y) => '0000'.substr(y.length) + y);
                 return [d.month + 1, d.day, yearStr].join('/');
               },
-              parseDate: (text) => {
+              parseDate(text) {
                 const parts = text.split('/');
                 const today = new Date();
                 let date,
                   month = today.getMonth(),
                   year = today.getFullYear();
 
                 if (parts.length === 3) {
+                  month = parseInt(parts[0]) - 1;
+                  date = parseInt(parts[1]);
                   year = parseInt(parts[2]);
                   if (parts[2].length < 3 && year >= 0) {
-                    year += year < 50 ? 2000 : 1900;
+                    const usedReferenceDate = this.referenceDate ? new Date(this.referenceDate) : new Date();
```

Comment:
We should probably reuse the `_parseDate` function here to parse the reference date:
https://github.com/vaadin/web-components/blob/427527c27c4b27822d61fd41d38d7b170134770b/packages/date-picker/src/vaadin-date-picker-mixin.js#L609-L622

Currently when configuring a reference date in an invalid format (e.g. `2050-1-1`) the `i18n.parseDate` function just silently fails when entering a year with two digits.

Note that we have to extract the function above, maybe to `vaadin-date-picker-helper.js`, so that it can be used from `i18n.parseDate`.